### PR TITLE
#46 - add Ballp and docs

### PR DESCRIPTION
**Pull request by mforets**

### CODE REVIEW

**Comment by schillic** [comment]
Can we remove `Ball2` then?

**Comment by mforets** [comment]
i think ball2 is sufficiently "special" to keep it separated; this general p-norm ball is more rarely used. note that it falls back to ball2 whenever `p=2` is chosen

**Comment by mforets** [comment]
i didn't add yet unit tests for this file. some computation by hand can be done for `p=1`.

**Comment by schillic** [comment]
The only difference is that we do not store `p` and the computation of `σ`.
Why is `σ` different?
Also I think it will crash if `||d|| = 0``. `Ball2` has a case split for this.

**Comment by schillic** [comment]
Is `p::Real` correct? Does not make sense to me...

**Comment by mforets** [comment]
>  Why is σ different?

in [Le Guernic's thesis, Appendix A page 125](https://tel.archives-ouvertes.fr/tel-00422569v1/document) there is the proof for the p-norm in the special case `c=0` and `r=1`. as written in the new docstring of `Ballp`, i used an affine transformation to extend it for any `c` and `r`.

>  Also I think it will crash if ||d|| = 0. Ball2 has a case split for this.

yes, i can add a case for this (we should return 0 as well).

**Comment by mforets** [comment]
shoud fix `p<= 1`. thanks :)

**Comment by mforets** [comment]
actually i'll go for creating a new type `Ball1`. the reason is that this one is "special" similarly to the 2 and oo cases.

File: docs/make.jl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 using Documenter, LazySets
 
 makedocs(
-#   doctest = false,  # uncomment locally for skipping doctests (saves time!)
```
Comment (schillic) [inline docs/make.jl]: I would prefer to keep the `false` uncommented.
By default it is active, so ne need to write this explicitly.

Comment (mforets) [reply docs/make.jl]: yes, this was a distraction, thanks

Comment (mforets) [reply docs/make.jl]: well, actually i had written `true` in the right hand side :)

Comment (schillic) [reply docs/make.jl]: You need to set it to `false`;)

**Pull request merged**